core/src/main/scala/org/apache/spark/util/collection/ExternalSorter.scala

```diff
@@ -710,7 +711,7 @@ private[spark] class ExternalSorter[K, V, C](
             serializerManager,
             serInstance,
             blockId,
-            context.taskMetrics().shuffleWriteMetrics,
```

Comment:
Hmm, the `metrics` given at `getWriter` looks like also coming from `context.taskMetrics().shuffleWriteMetrics`, isn't it looking the same?

Reply:
It may be `SQLShuffleWriteMetricsReporter`, which may not be the same as `context.taskMetrics().shuffleWriteMetrics`.


https://github.com/apache/spark/blob/9e12bd7f2b40051f7540b14d85a3ea1f3e41f6a1/core/src/main/scala/org/apache/spark/shuffle/ShuffleWriteProcessor.scala#L51-L56

https://github.com/apache/spark/blob/9e12bd7f2b40051f7540b14d85a3ea1f3e41f6a1/sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/ShuffleExchangeExec.scala#L449-L455